File: backend/demo/src/main/java/com/example/demo/dtos/ContratAssuranceDTO.java
package com.example.demo.dtos;

import lombok.Data;
import java.util.Date;

@Data
public class ContratAssuranceDTO {

    private Long id;
    private Date dateSouscription;
    private Date dateValidation;

    private String statut;

    private double montantCotisation;
    private int dureeContrat;
    private double tauxCouverture;

    private Long clientId;

    private String typeContrat;
}

File: frontend/frontend/src/app/model/ContratAssuranceDTO.ts
export interface ContratAssuranceDTO {
  id?: number;
  dateSouscription?: Date | string;
  dateValidation?: Date | string;
  statut: string;
  montantCotisation: number;
  dureeContrat: number;
  tauxCouverture: number;
  clientId?: number;
  typeContrat: string;
}
